src/repositories/searchResult.ts

```diff
@@ -18,13 +20,18 @@ export const createSearchResult = async (
 export const updateSearchResult = async (args: Prisma.SearchResultUpdateManyArgs) =>
   dbClient.searchResult.updateMany(args);
 
-export const listSearchResult = async (userId: number) => {
-  const data = await dbClient.searchResult.findMany({
-    where: {
-      userId,
-    },
-    select: searchResultScraping,
+export const listSearchResult = async (
+  query: { [key: string]: any } = {},
+  paginationParams: PaginationQueryParams
+) =>
+  dbClient.searchResult.findMany({
+    where: query,
+    skip: paginationParams.offset,
+    take: paginationParams.limit,
   });
 
-  return data;
-};
+export const countSearchResult = async (query: { [key: string]: any } = {}) =>
+  dbClient.searchResult.count({ where: query });
+
+export const listAllPendingSearchResult = async (userId: number) =>
```

Comment:
How about making it more generic (so that it can handle any status and setting a default):

```suggestion
export const listSearchResultByStatus = async (userId: number, status: string = SearchResultStatus.PENDING) =>
```

Reply:
Fixed in https://github.com/nimblehq/ic-nextjs-prabesh-dahal/pull/75/commits/8dbebf789e17df5f983c903fad85b20f56d1c482